src/main/java/org/checkerframework/checker/noliteral/qual/MaybeConstant.java

```diff
@@ -0,0 +1,36 @@
+package org.checkerframework.checker.noliteral.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.DefaultInUncheckedCodeFor;
+import org.checkerframework.framework.qual.LiteralKind;
+import org.checkerframework.framework.qual.QualifierForLiterals;
+import org.checkerframework.framework.qual.SubtypeOf;
+import org.checkerframework.framework.qual.TypeUseLocation;
+
+/**
+ * The top type for the NoLiteral Checker's type system.
+ *
+ * <p>This type represents an expression that might contain a literal, like 5 or {0xa, 0xb}. It is
+ * inferred for all manifest literals in the program.
```

Comment:
"It is inferred for" is an implementation detail.  Just say those literals have the type, or omit it since it is obvious from the first sentence.